### PR TITLE
Use attrs instead of NamedTuple to fix missing call to validate

### PR DESCRIPTION
**Pull request by rcoh**
Using attrs opened up a can of worms, but probably a worthwhile one. This PR switchs from NamedTuple to attrs to support post-init validators. But attrs uses uuid, and uuid uses libuuid. And loading libuuid doesn't work on alpine which breaks nuitka. 

So...I also had to remove nuitka from the docker image.

### CODE REVIEW

**Comment by brendongo** [comment]
Closes https://github.com/returntocorp/semgrep/issues/763

File: semgrep/Pipfile [hunk 1]
```diff
@@ -11,6 +11,7 @@ pytest-snapshot = "*"
 
 [packages]
 semgrep = {editable = true,path = "."}
+attrs = "*"
```
Comment (brendongo) [inline semgrep/Pipfile]: You probably want to add this to setup.py

Comment (rcoh) [reply semgrep/Pipfile]: oh good call.

**Comment by rcoh** [comment]
build failure appears to be an actual nuitka problem

**Comment by rcoh** [comment]
Docker failure is now spurious -- I should really stop making external PRs or we should remove the push job for PRs.

**Review by brendongo — APPROVED** [review]

**Pull request merged**